backend/models/userModel.js

```diff
@@ -0,0 +1,46 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcrypt");
+
+
+const userSchema = mongoose.Schema(
+  {
+    name: {
+      type: String,
+      required: true,
+    },
+    email: {
+      type: String,
+      required: true,
+      unique: true,
+    },
+    password: {
+      type: String,
+      required: true,
+    },
+  },
+  {
+    // tells mongoose to assign createdAt and updatedAt fields to the user schema
+    timestamps: true,
+  }
+);
+
+userSchema.methods.matchPassword = async function (enteredPassword) {
+  return await bcrypt.compare(enteredPassword, this.password);
+};
+
+// Before we perform the command save, the async function will run
+userSchema.pre("save", async function (next) {
+  // This function will automatically run pre save
+
+  if (!this.isModified("password")) {
+    next();
+  }
+
+  const salt = await bcrypt.genSalt(10);
+  this.password = await bcrypt.hash(this.password, salt); // hasing the provided password
```

Comment:
I believe the comment above applies here as well

Reply:
Refer to this link:; https://attacomsian.com/blog/nodejs-password-hashing-with-bcrypt#:~:text=Bcrypt%20provides%20both%20asynchronous%20and,other%20incoming%20requests%20or%20events.

Reply:
This makes sense!